### DISCUSSION TITLE
Pseudo support for typescript extensions

### DISCUSSION THREAD

**Discussion by benosman** [directus/directus#15718] [Feature Requests]
I know this question has been asked a couple of times and decided against for sensible reasons, and instead directus-extension suggested.

#8324 #9233

However, being able to use typescript extensions for the backend (hooks, endpoints, migrations, operations) is something that was very important for our use case to allow easier testing and debugging.

I have actually got this working well and reliably for our dev environments and then on production it all gets compiled to javascript so runs normally. 

Having set this up, I do understand that creating this as a one-fits-all solution is not possible, as it needs extra tooling and configuration. Surprisingly most of it can be done without modifying directus, basically just adding some ts-node-dev glue to the package.json file and the thinnest of wrappers around the start process and the cli.

There are however a couple of lines of code in [@directus/shared/src/utils/node/get-extensions.ts](https://github.com/directus/directus/blob/cb3d775c4e56313c0643ef367412b0c99e33eec4/packages/shared/src/utils/node/get-extensions.ts#L118-L135)
where the `.js` extension is hardcoded.

Just to be clear, I'm not asking for any changes to the built in functionality of directus, but I wondered if there would be any willingness to allow advanced configurations to override these file extensions?

I would be happy to talk through more specifics if there is any interest.

**Comment by rijkvanzanten** [comment]
cc @nickrum

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
